### PR TITLE
Odd folding inputs correction

### PR DESCRIPTION
**Pull request by daniel-mills-cqc**
This PR corrects the inputs to odd_folding, and adds additional testing which would catch incorrect input formatting.

### CODE REVIEW

File: qermit/zero_noise_extrapolation/zne.py [hunk 1]
```diff
@@ -47,7 +46,7 @@ class Folding(Enum):
     # TODO: Understand why self is needed etc
```
Comment (sjdilkes) [inline qermit/zero_noise_extrapolation/zne.py:46]: can we get rid of some of these TODO?

File: qermit/zero_noise_extrapolation/zne.py [hunk 2]
```diff
@@ -539,7 +542,7 @@ def task(
         for experiment in mitex_wire:
 
             # Apply the necessary folding method
-            zne_circ = _folding_type(experiment.AnsatzCircuit.Circuit, noise_scaling, _allow_approx_fold)  # type: ignore
+            zne_circ = _folding_type(experiment.AnsatzCircuit.Circuit, noise_scaling, _allow_approx_fold=_allow_approx_fold)  # type: ignore
 
             # TODO: This additional compilation pass may result in the circuit noise being
```
Comment (sjdilkes) [inline qermit/zero_noise_extrapolation/zne.py:547]: This todo?

**Review by sjdilkes — CHANGES_REQUESTED** [review]
Annoying request only, but if the TODO statements I mentioned can be removed easily that would be fab.

File: qermit/zero_noise_extrapolation/zne.py (hunk 2 above)
Comment (daniel-mills-cqc) [reply qermit/zero_noise_extrapolation/zne.py:546]: This one is still todo I think

**Pull request merged**